### PR TITLE
added docker & converted to session for token

### PR DESCRIPTION
**Pull request by stottidfe**
https://dfe-secureaccess.atlassian.net/browse/DSI-6627 added token id to session redis cache
https://dfe-secureaccess.atlassian.net/browse/DSI-6805 added necessary resources to run docker

### CODE REVIEW

File: src/infrastructure/oidc/index.js [hunk 1]
```diff
@@ -115,7 +115,7 @@ const init = async (app) => {
       }
 
       if(allUserServices && allUserServices.roles){
-        const { roles } = allUserServices;
+        const roles = allUserServices.roles.sort((a, b) => a.name.localeCompare(b.name, 'es', {sensitivity: 'base'}));
```
Comment (malikimran34) [inline src/infrastructure/oidc/index.js]: Was this expected changes

File: src/index.js [hunk 2]
```diff
@@ -30,7 +42,12 @@ if (config.hostingEnvironment.applicationInsights) {
   appInsights.setup(config.hostingEnvironment.applicationInsights).start();
 }
 
+
 const init = async () => {
+  localStorage.init({
```
Comment (malikimran34) [inline src/index.js:47]: Is local storage necessary to add not sure how big the object is expected to be stored in local storage?

File: src/infrastructure/oidc/index.js (hunk 1 above)
Comment (stottidfe) [reply src/infrastructure/oidc/index.js]: I think this crept in on the back of a pair programming session with Adam I can remove if you prefer?

File: src/index.js (hunk 2 above)
Comment (stottidfe) [reply src/index.js:47]: After inspecting previously added functionality in the other implementations of this in manage and services this was added to the init function and I thought it was required if not happy to remove it?

**Review by malikimran34 — APPROVED** [review]

**Pull request merged**